ymir.sh

```diff
@@ -16,9 +16,52 @@ FIELD_LABEL_TOOL_TOKEN='LABEL_TOOL_TOKEN'
 FIELD_LABEL_TOOL_LS='label_studio'
 FIELD_LABEL_TOOL_LF='label_free'
 ENV_FILE='.env'
-
+FIELD_SERVER_RUNTIME='SERVER_RUNTIME'
 FIELD_DEPLOY_MODULE_HOST_PORT='DEPLOY_MODULE_HOST_PORT'
 
+check_docker_compose_version() {
+    MIN_DC_VER="1.29.2"
+    if ! command -v docker-compose &> /dev/null; then
+        echo "please install docker-compose ${MIN_DC_VER} or newer version."
+        exit
+    fi
+    if ! echo "$(docker-compose version --short) ${MIN_DC_VER}" | tr " " "\n" | sort -V | head -n 1 | grep -Eq "^${MIN_DC_VER}$"; then
+        echo "please upgrade docker-compose to ${MIN_DC_VER} or newer version."
+        exit
+    fi
+}
+
+check_docker_version() {
+    MIN_DOCKER_VER="20.10"
+    if ! command -v docker &> /dev/null; then
+        echo "please install docker ${MIN_DOCKER_VER} or newer version."
+        exit
+    fi
+    if ! echo "$(docker version --format '{{.Client.Version}}') ${MIN_DOCKER_VER}" | tr " " "\n" | sort -V | head -n 1 | grep -Eq "^${MIN_DOCKER_VER}$"; then
+        echo "please upgrade docker to ${MIN_DOCKER_VER} or newer version."
+        exit
+    fi
+}
+
+has_nvidia_gpu(){
+    if ! command -v nvidia-smi &> /dev/null; then
+        return 1
+    fi
+    if [ $(nvidia-smi -L | grep UUID:| wc -l) -eq 0 ]; then
+        return 1
+    fi
+    return 0
+}
+
+check_server_runtime(){
+    if cat ${ENV_FILE} | grep -oE "^${FIELD_SERVER_RUNTIME}=nvidia$"; then
+        if ! has_nvidia_gpu; then
+            echo "please make sure nvidia gpu is available when server runtime is nvidia."
```

Comment:
please make sure Nvidia driver is installed when server runtime is set to nvidia.